File: frontend/src/types/email-template.ts
export type EmailTemplateType =
  | 'invoice_send'
  | 'invoice_reminder_1'
  | 'invoice_reminder_2'
  | 'invoice_reminder_3'
  | 'credit_note'
  | 'document_send';

export interface EmailTemplate {
  id: string;
  template_type: EmailTemplateType;
  subject: string;
  body_html: string;
  language: string;
  is_default: boolean;
}

export interface UpdateEmailTemplate {
  subject: string;
  body_html: string;
}

export interface EmailTemplatePreview {
  rendered_subject: string;
  rendered_body_html: string;
}

File: backend/crates/konto-api/src/dto/email_template.rs
use serde::{Deserialize, Serialize};
use utoipa::ToSchema;

#[derive(Debug, Clone, Serialize, ToSchema)]
pub struct EmailTemplateResponse {
    pub id: String,
    pub template_type: String,
    pub subject: String,
    pub body_html: String,
    pub language: String,
    pub is_default: bool,
}

impl From<konto_db::entities::email_template::Model> for EmailTemplateResponse {
    fn from(m: konto_db::entities::email_template::Model) -> Self {
        Self {
            id: m.id,
            template_type: m.template_type,
            subject: m.subject,
            body_html: m.body_html,
            language: m.language,
            is_default: m.is_default,
        }
    }
}

#[derive(Debug, Deserialize, ToSchema)]
pub struct UpdateEmailTemplateRequest {
    pub subject: String,
    pub body_html: String,
}

#[derive(Debug, Serialize, ToSchema)]
pub struct EmailTemplatePreviewResponse {
    pub rendered_subject: String,
    pub rendered_body_html: String,
}
